python_modules/dagster/dagster/_core/definitions/repository_definition.py

```diff
@@ -1379,6 +1416,78 @@ def __call__(self, *args, **kwargs):
         return self
 
 
+class PendingRepositoryDefinition:
+    def __init__(
+        self,
+        name: str,
+        repository_definitions: Sequence[
+            Union[RepositoryListDefinition, "CacheableAssetsDefinition"]
+        ],
+        description: Optional[str] = None,
+        default_logger_defs: Optional[Mapping[str, LoggerDefinition]] = None,
+        default_executor_def: Optional[ExecutorDefinition] = None,
+    ):
+        self._repository_definitions = check.list_param(
+            repository_definitions,
+            "repository_definition",
+            additional_message="PendingRepositoryDefinition supports only list-based repository data at this time.",
+        )
+        self._name = name
+        self._description = description
+        self._default_logger_defs = default_logger_defs
+        self._default_executor_def = default_executor_def
+
+    @property
+    def name(self) -> str:
+        return self._name
+
+    def get_repository_load_data(self) -> RepositoryLoadData:
+        from dagster._core.definitions.cacheable_assets import CacheableAssetsDefinition
+
+        return RepositoryLoadData(
+            cached_data_by_key={
+                defn.unique_id: defn.get_cached_data()
+                for defn in self._repository_definitions
+                if isinstance(defn, CacheableAssetsDefinition)
+            }
+        )
+
+    def resolve(self, repository_load_data: RepositoryLoadData) -> RepositoryDefinition:
```

Comment:
re good stacks would put the meat of resolution in a private `_` prefixed method and have 2 distinct entry points ie  `warm_load(data)`  / `cold_load()`